model/simulationtests/surface_properties_lwr.rb

```diff
@@ -0,0 +1,179 @@
+
+s_sp.autocalculateGroundViewFactor
+s_sp.setGroundViewFactor(0.0)
+
+groundTempSch = OpenStudio::Model::ScheduleConstant.new(model)
+s_sp.setGroundTemperatureSchedule(groundTempSch)
+s_sp.resetGroundTemperatureSchedule
+
+tempSch1 = OpenStudio::Model::ScheduleRuleset.new(model)
+tempSch1.setName('Surface Temperature')
+tempSch1.defaultDaySchedule.addValue(OpenStudio::Time.new(0, 24, 0, 0), 10.0)
+tempSch1_july_to_dec_rule = OpenStudio::Model::ScheduleRule.new(tempSch1)
+tempSch1_july_to_dec_rule.setStartDate(OpenStudio::Date.new('July'.to_MonthOfYear, 1))
+tempSch1_july_to_dec_rule.setEndDate(OpenStudio::Date.new('December'.to_MonthOfYear, 31))
+tempSch1_july_to_dec_rule.daySchedule.addValue(OpenStudio::Time.new(0, 24, 0, 0), -10)
+
+# Add two groups
+# Group 1: the explicit SurroundingSurfaceGroup approach
+group1 = OpenStudio::Model::SurroundingSurfaceGroup.new('SurroundingSurface1', 0.5, tempSch1)
+s_sp.addSurroundingSurfaceGroup(group1)
+
+# Group 2: via the helper
+s_sp.addSurroundingSurfaceGroup('SurroundingSurface2', 0.3, tempSch1)
+
+###############################################################################
+#                              A P I    D E M O                               #
+###############################################################################
+
+# Demonstrate the api a bit more
+raise if s_sp.numberofSurroundingSurfaceGroups != 2
+
+# You can navigate up and down
+raise if localEnv.surfacePropertySurroundingSurfaces.get != s_sp
+raise if s_sp.surfacePropertyLocalEnvironment.get != localEnv
+
+# Note that the gorupIndex matches on Surrounding Surface name only
+raise if s_sp.surroundingSurfaceGroupIndex(group1).get != s_sp.surroundingSurfaceGroupIndex('SurroundingSurface1').get
+
+# Also note that API enforces uniqueness of Surrouding Surface Name
+# if I try to add another group with the same surface name, it overrides the
+# values
+s_sp.addSurroundingSurfaceGroup('SurroundingSurface2', 0.4, tempSch1)
+raise if s_sp.numberofSurroundingSurfaceGroups != 2
+raise if s_sp.surroundingSurfaceGroups[1].viewFactor != 0.4
+
+# Save the groups
+groups = s_sp.surroundingSurfaceGroups
+
+s_sp.removeSurroundingSurfaceGroup(1)
+raise if s_sp.numberofSurroundingSurfaceGroups != 1
+
+s_sp.addSurroundingSurfaceGroup('SurroundingSurface2', 0.4, tempSch1)
+s_sp.removeAllSurroundingSurfaceGroups
+raise if s_sp.numberofSurroundingSurfaceGroups != 0
+
+# Add back via the vector overload
+s_sp.addSurroundingSurfaceGroups(groups)
+raise if s_sp.numberofSurroundingSurfaceGroups != 2
+
+g1 = s_sp.surroundingSurfaceGroups[0]
+
+raise if g1.surroundingSurfaceName != 'SurroundingSurface1'
+raise if g1.viewFactor != 0.5
+raise if g1.temperatureSchedule != tempSch1
+
+g2 = s_sp.getSurroundingSurfaceGroup(1).get
+raise if g2.surroundingSurfaceName != 'SurroundingSurface2'
+raise if g2.viewFactor != 0.4
+raise if g2.temperatureSchedule != tempSch1
+
+###############################################################################
```

Comment:
Includng showcasing how to work with the extensible groups and the helper class `SurroundingSurfaceGroup` for the `SurfacePropertySurroundingSurfaces` class